File: app/src/main/java/com/starostinvlad/fan/BaseMVP/BasePresenter.java
package com.starostinvlad.fan.BaseMVP;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BasePresenter<T extends MvpView> implements MvpPresenter<T> {
    protected CompositeDisposable disposables = new CompositeDisposable();

    protected T view;

    @Override
    public void attachView(T mvpView) {
        view = mvpView;
    }

    @Override
    public void detachView() {
        view = null;
        disposables.dispose();
    }

}

File: app/src/main/java/com/starostinvlad/fan/BaseMVP/BasePresenter.kt
package com.starostinvlad.fan.BaseMVP

import io.reactivex.disposables.CompositeDisposable
import kotlin.Throws

abstract class BasePresenter<T : MvpView?> : MvpPresenter<T> {
    protected var disposables = CompositeDisposable()
    protected var view: T? = null
    override fun attachView(mvpView: T) {
        view = mvpView
    }

    override fun detachView() {
        view = null
        disposables.dispose()
    }
}